ParseLoginUI/src/main/java/com/parse/ParseQueryRecyclerViewAdapter.java

```diff
@@ -0,0 +1,611 @@
+
+    private Context context;
+
+    private List<T> objects = new ArrayList<>();
+
+    private Set<ParseQuery> runningQueries = Collections.newSetFromMap(new ConcurrentHashMap<ParseQuery, Boolean>());
+
+    // Used to keep track of the pages of objects when using CACHE_THEN_NETWORK. When using this,
+    // the data will be flattened and put into the objects list.
+    private List<List<T>> objectPages = new ArrayList<>();
+
+    private int currentPage = 0;
+
+    private Integer itemResourceId;
+
+    private boolean hasNextPage = true;
+
+    private QueryFactory<T> queryFactory;
+
+    private OnClickListener<T> onClickListener;
+
+    private List<OnQueryLoadListener<T>> onQueryLoadListeners = new ArrayList<>();
+
+    /**
+     * Constructs a {@code ParseQueryRecyclerViewAdapter}. Given a {@link ParseObject} subclass,
+     * this adapter will fetch and display all {@link ParseObject}s of the specified class,
+     * ordered by creation time.
+     *
+     * @param context
+     *          The activity utilizing this adapter.
+     * @param clazz
+     *          The {@link ParseObject} subclass type to fetch and display.
+     */
+    public ParseQueryRecyclerViewAdapter(Context context, Class<? extends ParseObject> clazz) {
+        this(context, ParseObject.getClassName(clazz));
+    }
+
+    /**
+     * Constructs a {@code ParseQueryRecyclerViewAdapter}. Given a {@link ParseObject} subclass,
+     * this adapter will fetch and display all {@link ParseObject}s of the specified class, ordered
+     * by creation time.
+     *
+     * @param context
+     *          The activity utilizing this adapter.
+     * @param className
+     *          The name of the Parse class of {@link ParseObject}s to display.
+     */
+    public ParseQueryRecyclerViewAdapter(Context context, final String className) {
+        this(context, new QueryFactory<T>() {
+            @Override
+            public ParseQuery<T> create() {
+                ParseQuery<T> query = ParseQuery.getQuery(className);
+                query.orderByDescending("createdAt");
+
+                return query;
+            }
+        });
+
+        if (className == null) {
+            throw new RuntimeException("You need to specify a className for the ParseQueryAdapter");
+        }
+    }
+
+    /**
+     * Constructs a {@code ParseQueryRecyclerViewAdapter}. Given a {@link ParseObject} subclass,
+     * this adapter will fetch and display all {@link ParseObject}s of the specified class, ordered
+     * by creation time.
+     *
+     * @param context
+     *          The activity utilizing this adapter.
+     * @param clazz
+     *          The {@link ParseObject} subclass type to fetch and display.
+     * @param itemViewResource
+     *        A resource id that represents the layout for an item in the AdapterView.
+     */
+    public ParseQueryRecyclerViewAdapter(Context context, Class<? extends ParseObject> clazz,
+                             int itemViewResource) {
```

Comment:
nit style: If it can fit on one line, good. If it can fit on two lines, all params on second line. If it can't fit on two lines, each param on a new line.

```java
public ParseQueryRecyclerViewAdapter(
    Context context, Class<? extends ParseObject> clazz, int itemViewResource) {
```